atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraUtils.java

```diff
@@ -37,4 +41,11 @@ private CassandraUtils() {}
             CassandraKeyValueServiceConfig config) {
         return client -> client.describe_ring(config.getKeyspaceOrThrow());
     }
+
+    public static AtlasDbDependencyException wrapInIceForDeleteOrRethrow(RetryLimitReachedException ex) {
+        if (ex.suppressed(UnavailableException.class) || ex.suppressed(InsufficientConsistencyException.class)) {
+            throw new InsufficientConsistencyException("Deleting requires all Cassandra nodes to be available.", ex);
+        }
```

Comment:
This happens if at any point we got one of these exceptions. So we might not be basing this on the latest state, but I think this is fine